doc/userDocs/usersGuide.md

```diff
@@ -1418,6 +1519,93 @@ As a result, when using node labels it is recommended that you label all of your
 worker nodes with either `wvrole=driver` or `wvrole=sut`.  This will give you the 
 best control over pod placement.
 
+### Namespace Customization <a name="advanced-namespaces"></a>
+
+#### Customizing the Generation of Namespace Names
+
+Weathervane generates the namespace names using three components:
+* A namespace prefix, which defaults to **auction**.
+* A unique identifier, which is __w1__ for the drivers and __w1i*n*__ for each application 
+  instance __*n*__.
+* A namespace suffix, which is blank by default.
+
+It is possible to change the strings used for the prefix and suffix using the parameters 
+`namespacePrefix` and `namespaceSuffix`.  These parameters are placed in the configuration 
+file within the definition of the Kubernetes cluster.
+
+For example, the following configuration file excerpt will use namespace names of the 
+form __appw1i*n*small__ for the cluster named *appCluster* and __driverw1small__ for 
+the cluster named *driverCluster*.
+
+```json
+"kubernetesClusters" : [ 
+  { 
+    "name" : "appCluster", 
+    "kubeconfigFile" : "/root/.kube/config",
+    "kubeconfigContext" : "cluster-context-1",
+    "namespacePrefix" : "app",
+    "namespaceSuffix" : "small",
+  },
+  { 
+    "name" : "driverCluster", 
+    "kubeconfigFile" : "/root/.kube/config",
+    "kubeconfigContext" : "cluster-context-2",
+    "namespacePrefix" : "driver",
+    "namespaceSuffix" : "small",
+  },
+],
+```
+
+#### Specifying Namespace Names
+
+It is possible to use the `namespaces` parameter to provide a list of namespace names for each 
+Kubernetes cluster.  Weathervane will use namespace names from that list rather than 
+generating the namespace names. Any values specified for `namespacePrefix` and `namespaceSuffix` 
+will be ignored.
+
+The following configuration file excerpt shows how to specify a list of namespace names. In 
+this example Weathervane will use *devns1* and *devns2* as the names for the workload driver 
+and application namespaces. 
+
+```json
+"kubernetesClusters" : [ 
+  { 
+    "name" : "k8sCluster", 
+    "kubeconfigFile" : "/root/.kube/config",
+    "kubeconfigContext" : "cluster-context-1",
+    "namespaces" : ["devns1", "devns2",],
+  },
+],
+```
+When specifying namespace names, you must provide at least one namespace name for each 
+application instance you plan to run on the cluster, plus one namespace name for the 
+workload drivers.  This means that if you are running the drivers and applications 
```

Comment:
I would switch the order of these, since the driver namespace is allocated first? (per the comment in the other pr) 
workload drivers namespace then application instance namespaces.

Reply:
Agreed.  I have reworded.